File: YTMSiparis/PremiumUyg/pyhton/Kullanici.py
class Kullanici:
    def __init__(self):
        self.id = None  # id için bir başlangıç değeri
        self.isim = None  # isim için bir başlangıç değeri
        self.mail = None  # mail için bir başlangıç değeri
        self.urunler = []  # ürünler listesi

    def get_toplam_tutar(self, fiyat):
        raise NotImplementedError("Bu metot, alt sınıflar tarafından geçersiz kılınmalıdır.")

    @property
    def isim(self):
        return self.isim

    @property
    def urunler(self):
        return self.urunler

    def get_urun(self, index):
        return self.urunler[index]

    def add_urun(self, urun):
        self.urunler.append(urun)

File: YTMSiparis/PremiumUyg/c#/OCP/OCP/Kullanici.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace OCP
{
    public abstract class Kullanici
    {
        protected int id;
        protected string isim;
        protected string mail;
        protected List<Urun> urunler;

        public Kullanici()
        {
            urunler = new List<Urun>();
        }
        public abstract double GetToplamTutar(double fiyat);
        public string Isim
        {
            get { return isim; }
        }
        public List<Urun> Urunler { get { return urunler; } }
        public Urun GetUrun(int index)
        {
            return Urunler[index];
        }
        public void AddUrun(Urun urun)
        {
            Urunler.Add(urun);
        }
    }
}

